lib/rex/post/meterpreter/client_core.rb

```diff
@@ -768,9 +768,16 @@ def negotiate_tlv_encryption(timeout: client.comm_timeout)
       response = client.send_request(request, timeout)
       key_enc = response.get_tlv_value(TLV_TYPE_ENC_SYM_KEY)
       key_type = response.get_tlv_value(TLV_TYPE_SYM_KEY_TYPE)
-
+      is_weak_key = false
       if key_enc
-        sym_key = rsa_key.private_decrypt(key_enc, OpenSSL::PKey::RSA::PKCS1_PADDING)
+        key_dec_data = rsa_key.private_decrypt(key_enc, OpenSSL::PKey::RSA::PKCS1_PADDING)
+
+        if key_dec_data.length == 17 || key_dec_data.length == 33
+          sym_key = key_dec_data[0, key_dec_data.length - 1]
+          is_weak_key = key_dec_data[key_dec_data.length - 1] == "\x01"
```

Comment:
It's probably being paranoid, but let's invert the logic here so it's more tamper-resistant. This way we'll warn on anything that's unexpected, e.g. \x02, \x03 (which should never happen).

```suggestion
          is_weak_key = key_dec_data[key_dec_data.length - 1] != "\x00"
```